src/styles/index.scss

```diff
@@ -1,3 +1,85 @@
+$day-width: 100px;
+$day-height: 100px;
+$day-gap: 1px;
+$day-color: #eee;
+$day-color-hover: #ffbfcb;
+$day-border: 1px solid #000;
+$day-day: (
+  'mon' 0,
+  'tue' $day-width + $day-gap,
+  'wed' 2 * ($day-width + $day-gap),
+  'thu' 3 * ($day-width + $day-gap),
+  'fri' 4 * ($day-width + $day-gap),
+  'sat' 5 * ($day-width + $day-gap),
+  'sun' 6 * ($day-width + $day-gap)
+);
+
 body {
   margin: 0;
 }
+
+.container {
+  width: 100vw;
+  height: 100vh;
+
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+
+.calendar {
+  width: $day-width * 7 + 6;
+  display: flex;
+  flex-wrap: wrap;
+  gap: $day-gap;
+
+  &__day {
+    box-sizing: border-box;
+
+    width: $day-width;
+    height: $day-height;
+
+    border: $day-border;
+
+    background-color: $day-color;
+
+    position: relative;
+
+    cursor: pointer;
+
+    transition:
+      transform 0.5s,
+      background-color 0.5s;
+  }
+
+  &__day:hover {
+    transform: translateY(-20px);
+    background-color: $day-color-hover;
+  }
+
+  @for $i from 29 through 31 {
+    &--month-length-#{$i} {
+      &__day:nth-child(n + #{$i + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  @for $i from 1 through 31 {
+    &__day:nth-child(#{$i})::before {
+      content: '#{$i}';
+      position: absolute;
+      top: 50%;
+      left: 50%;
+      transform: translate(-50%, -50%);
+      font-family: Arial, sans-serif;
+      font-size: 30px;
```

Comment:
- Don't use loops for styles that stay the same
- To center an element it's better to use `flex` instead of absolute positioning